crates/cxx-qt-lib/src/qml/qqmlapplicationengine.rs

```diff
@@ -88,13 +97,30 @@ mod ffi {
 
 use crate::QQmlEngine;
 use core::pin::Pin;
+use cxx_qt::Upcast;
 
 pub use ffi::QQmlApplicationEngine;
 
+impl Upcast<QQmlEngine> for QQmlApplicationEngine {
+    unsafe fn upcast_ptr(this: *const Self) -> *const QQmlEngine {
+        ffi::upcast_qqmlapplication_engine(this)
+    }
+
+    #[cfg(cxxqt_qt_version_at_least_6)]
+    unsafe fn from_base_ptr(base: *const QQmlEngine) -> *const Self {
+        ffi::downcast_qqml_engine(base)
+    }
+
+    #[cfg(cxxqt_qt_version_major = "5")]
+    unsafe fn from_base_ptr(_base: *const QQmlEngine) -> *const Self {
+        std::ptr::null()
+    }
```

Comment:
why can't Qt 5 downcast ? It supposedly has a virtual on the deconstructor of QQmlEngine